### DISCUSSION TITLE
json parameters - Query REST API - DONE

### DISCUSSION THREAD

**Discussion by casualmatt** [Budibase/budibase#1688] [Ideas]
hello, 
i'm new with nocode, and coding in general. 

I want to use as external data-source a rest endpoint. 
Example: 

GET pizzeria.it/pizza
and i will get:
```
{
   "pizza":[
      {
         "id":1,
         "name":"margherita",
         "cost":"10"
      },
      {
         "id":2,
         "name":"napoli",
         "cost":"10"
      }
   ],
   "meta":{
      "pagination":{
         "page":3,
         "per_page":25,
         "previous_page":2,
         "next_page":4,
         "last_page":4,
         "total_entries":100
      }
   }
}
```
budibase ask me the schema for the information, what can i put to have a list of my pizza?

**Comment by shogunpurple** [comment]
Hi @casualmatt - right now budibase expects data back from the API as an array of objects in order to build a schema properly. The ability to map the data coming back from the REST API to the format required for a schema has been raised here:

https://github.com/Budibase/budibase/discussions/1604

We will implement this over the coming weeks.

**Reply by timchkor** [reply]
Great.. we're waiting for this feature too.

**Reply by timchkor** [reply]
@shogunpurple any update on this feature? Thx!

**Reply by shogunpurple** [reply]
@timchkor we still haven’t got round to it yet, but it’s on the roadmap.

In the meantime, you should be able to do this in your data provider using handlebars operators such as ‘pluck’ on your array.

**Reply by karimsiam** [reply]
would you elaborate more on using handlebars with the Data Provider

**Reply by casualmatt** [reply]
@shogunpurple do you have some documentation or some example on how to do what you suggest?

**Reply by shogunpurple** [reply]
You can now do this with transformers - which are the easiest way to map data coming back from your data sources:

Docs here: https://docs.budibase.com/building-apps/data/transformers
